### ISSUE TITLE
Whether empty samples can be added for training PTPCG

### ISSUE THREAD

**Issue by TinaGioro** [Spico197/DocEE#38]
** Idea sharing **
While sharing what you want to do, make sure to protect your ideas.

** Problems **
If you have any questions about event extraction, make sure you have read the latest papers or searched on the Internet.

** Others **
Other things you may want to share or discuss.

Hello Spico, I'm reproducing your paper and use your online demo. I find a new event will be misclassified. So Can help model classify the event list as null if I add some empty samples. like this:
{text:'I love China',"event_list:"[]

**Comment by Spico197** [comment]
Hi, there. Thanks for the comments.
Since there are no negative samples in the training set, the model cannot well predict N/A event.
Besides, documents without event instances would be removed during pre-processing (for DuEE-fin only).
You can absolutely add an additional N/A event and negative samples with some code adaptation to provide a more accurate N/A instance prediction.

**Comment by TinaGioro** [comment]
Thanks for replying. I find another problem about what if there are more than one arguments in an argument role. Does the model take that into account?

**Comment by Spico197** [comment]
> Thanks for replying. I find another problem about what if there are more than one arguments in an argument role. Does the model take that into account?

Yes, PTPCG can predict more than one argument for each role. This is a common practice when we participated in LIC'21 (DuEE-fin):
- your data can be formatted as something like below:
```
"recguid_eventname_eventdict_list": [
  [
    0,
    "EquityPledge",
    {
      "Pledger": ["李云春"],
      "PledgedShares": ["40789194股", "123123股"],
      "Pledgee": ["招商证券股份有限公司"],
      "TotalHoldingShares": ["162103218股"],
      "TotalHoldingRatio": ["11.55%"],
      "TotalPledgedShares": ["86304393股"],
      "StartDate": ["2015年5月7日"],
      "EndDate": ["2016年5月6日"],
      "ReleasedDate": ["2016年5月6日"]
    }
  ]
]
```
- make code adaptations to fit such a format (e.g. 1) write new features in data helper to keep multi-arg roles ; 2) multi-hot encoding rather than one-hot when calculating role's classification loss)
- *important:* set `unique_role=False` when predicting argument roles: `predict_span_role(batch_span_emb, unique_role=False)`

However, since the other baselines (Doc2EDAG, GIT) are designed on ChFinAnn, they cannot predict many arguments for one role in a single event instance. To make fair comparisons with these systems, we set PTPCG to only predict one argument each time (`unique_role=True`), and publish the results in our paper.

**Comment by TinaGioro** [comment]
Thanks for replying. I am trying to modify the code.

**Comment by xxllp** [comment]
> Thanks for replying. I am trying to modify the code.

不知道是否有改好可以跑的代码

**Comment by BEILOP** [comment]
感谢回答，我的能力比较弱，还是有些懵懂。我做了以下几点：1、把数据转换为您所展示的一个论元多角色和一个角色多论元。2、set unique_role=False。3、使用EventTableForSigmoidMultiArgRel替换了EventTableForArgRel。4、role_classificiation的loss计算我看是binary_cross_entropy，好像不需要替换。而logits是log_softmax,我换成了sigmoid。5、正在看dee_helper可能需要修改的地方。

**Comment by BEILOP** [comment]
请问是否有疏漏的地方

**Comment by Spico197** [comment]
> 请问是否有疏漏的地方

看起来可以，如果后面遇到新的问题，欢迎继续讨论。
